kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -283,116 +311,175 @@ define([
+         *    "Searching..." or "Loading..." strings as temporary/unselectable options.
+         * 3. (on widget load), an object with id, text, and nothing else should just return
+         *    the text?
+         * See https://select2.org/dropdown#templating for more details.
          */
         function formatObjectDisplay(retObj) {
             if (!retObj.id) {
                 return retObj.text || '';
             } else if (dataSource === 'ftp_staging') {
-                if (!retObj.id) {
-                    return $(
-                        div(
-                            {
-                                class: `${baseCssClass}_display`,
-                            },
-                            StringUtil.escape(retObj.text)
-                        )
-                    );
-                }
-                return $(
-                    div([
-                        span(
-                            {
-                                class: `${baseCssClass}_display__filepath`,
-                            },
-                            [StringUtil.escape(retObj.subdir), b(StringUtil.escape(retObj.name))]
-                        ),
-                        div(
-                            {
-                                class: `${baseCssClass}_display__indent`,
-                            },
-                            [
-                                'Size: ' + StringUtil.readableBytes(retObj.size) + '<br>',
-                                'Uploaded ' + TimeFormat.getTimeStampStr(retObj.mtime, true),
-                            ]
-                        ),
-                    ])
-                );
+                return formatFtpStagingDisplay(retObj);
             } else {
-                const replacer = function (_match, p1) {
-                    return retObj[p1];
-                };
                 let formattedString;
-                if (ddOptions.description_template) {
-                    // use slice to avoid modifying global description_template
-                    formattedString = ddOptions.description_template
-                        .slice()
-                        .replace(/{{(.+?)}}/g, replacer);
-                } else {
-                    formattedString = JSON.stringify(retObj);
+                // if we have a template and at least one item to fill the template with,
+                // do the formatted template
+                if (
+                    ddOptions.description_template &&
+                    [...model.descriptionFields].some((key) => key in retObj)
+                ) {
+                    formattedString = formatDescriptionTemplate(retObj);
                 }
+                // otherwise, if we have a text use that. And if not, use the id as a last resort.
+                else {
+                    formattedString = retObj.text || retObj.id;
+                }
+                // and squash the result in a jQuery object with a div.
```

Comment:
Select2 wants either a string (which will be put in as a textContent field to avoid XSS things) or a jquery element. We do our own escaping with `util/string`, so there should be less danger.